curated_transformers/tests/generation/test_falcon.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+import torch
+
+from curated_transformers.generation.config import (
+    GreedyGeneratorConfig,
+    SampleGeneratorConfig,
+)
+from curated_transformers.generation.falcon import FalconGenerator
+
+from ..conftest import GPU_TESTS_ENABLED
+
+
+@pytest.fixture(scope="module")
+def falcon_generator():
+    return FalconGenerator.from_hf_hub(
+        name="tiiuae/falcon-7b-instruct", device=torch.device("cuda", index=0)
+    )
+
+
+@pytest.mark.veryslow
+@pytest.mark.skipif(not GPU_TESTS_ENABLED, reason="requires GPU")
+def test_generate_deterministic(falcon_generator):
+    prompts = [
+        "What is the Rust programming language?",
+        "What is spaCy?",
+    ]
+    assert falcon_generator(prompts, config=GreedyGeneratorConfig()) == [
```

Comment:
The test will look less noisy (and more debuggable) if we extracted the expected outputs to a variable and compared the outputs with it (`expected = expected[::-1]` for the reversed version).

Reply:
That's a great idea! Changed the Dolly tests in the same way.